### PR TITLE
Remove DeckGLOverlay in examples

### PR DESCRIPTION
**Pull request by ibgreen**
#### Background

For #1835 (Simplify Source Code of Website Examples)

#### Change List
- See checklist in issue

### CODE REVIEW

File: website/src/components/demos/geojson-demo.js [hunk 1]
```diff
@@ -124,7 +128,8 @@ export default class GeoJsonDemo extends Component {
 
     return (
       <div>
-        <GeoJsonOverlay viewport={viewport}
+        <App
+          viewport={viewport}
```
Comment (Pessimistress) [inline website/src/components/demos/geojson-demo.js]: No longer used?

File: examples/website/geojson/app.js [hunk 2]
```diff
@@ -11,56 +11,75 @@ const MAPBOX_TOKEN = process.env.MapboxAccessToken; // eslint-disable-line
...
-  }
+const DEFAULT_COLOR_SCALE = r => [r * 255, 140, 200 * (1 - r)];
 
-  _resize() {
-    this._onViewportChange({
-      width: window.innerWidth,
-      height: window.innerHeight
-    });
+export class App extends Component {
+  static get defaultViewport() {
```
Comment (Pessimistress) [inline examples/website/geojson/app.js]: Is this still needed?

Comment (ibgreen) [reply examples/website/geojson/app.js]: I believe that there is a viewport transition functionality in the "DemoLauncher". I think it is the one piece of existing functionality that I didn't maintain. 

I will take another look, assuming we need to keep this for now.

Comment (Pessimistress) [reply examples/website/geojson/app.js]: I don't think we have to keep the transition effect.

**Comment by Pessimistress** [comment]
My only concern about this change is the new component hierarchy (deck as the event handler and react-map as the child). I still consider the controller and views feature incomplete. I'd hold on using this pattern in primary examples till v6.

**Pull request merged**